user-service/src/controllers/userController.ts

```diff
@@ -0,0 +1,56 @@
+import { Request, Response } from "express";
+import bcrypt from "bcrypt";
+import User from '../models/userModel';
+import generateTokenAndSetCookie from '../lib/generateToken';
+import mongoose from "mongoose";
+
+export async function createUser(req: Request, res: Response) {
+  try {
+    const { username, email, password } = req.body;
+
+    // sanity check
+    const emailRegex = /^[^\s@]+@[^\s@]+\.[^\s@]+$/;
```

Comment:
can leave a comment to show what this regex means?

Reply:
<img width="728" alt="image" src="https://github.com/user-attachments/assets/be5f074b-1d13-4248-87d8-d8d00b399221">

hope this helps @tiongMax

Reply:
Ok

Reply:
ok done.